### PR TITLE
Add duplicate bookmark button

### PR DESCRIPTION
**Pull request by leonunix**
Sometime we have some servers with the same username/password only ip address is different.
So this feature is very convenient to do this.

### CODE REVIEW

File: src/client/components/setting-panel/tree-list.jsx [hunk 1]
```diff
@@ -23,7 +24,9 @@ import copy from 'json-deep-copy'
 import onDrop from './on-tree-drop'
 import Search from '../common/search'
 import Btns from './bookmark-transport'
+
 import './tree-list.styl'
+import time from '../../common/time';
```
Comment (zxdong262) [inline src/client/components/setting-panel/tree-list.jsx]: Make sure you run `npm i && npm run lint` before submit, you will get format tip from console if there are any lint errors.

File: src/client/components/setting-panel/tree-list.jsx [hunk 2]
```diff
@@ -357,6 +360,45 @@ export default class ItemListTree extends React.PureComponent {
...
+
+    const {
+      bookmarkGroups
+    } = this.props
+
+    const groupid = this.findBookmarkGroupId(bookmarkGroups, item.id)
+    this.props.store.modifier({
+      currentBookmarkGroupId: groupid
+    })
+    this.props.onClickItem(newbookmark)
```
Comment (zxdong262) [inline src/client/components/setting-panel/tree-list.jsx]: Duplicate should works as one click to duplicate it at once, after click duplicate button, there should be another item appears in bookmarks list, with the title `copy:{old title}` or `copy:{old host}`(if no title), then go into edit this duplicated item mode, user can edit it and save.

**Comment by zxdong262** [comment]
@leonunix Thank for the PR, but requires same change on how it works and code format,  I think it is a good idea to have a duplicate button.

File: src/client/components/setting-panel/tree-list.jsx (hunk 1 above)
Comment (leonunix) [reply src/client/components/setting-panel/tree-list.jsx]: oh. It is auto input by ide. I will remove it.

File: src/client/components/setting-panel/tree-list.jsx (hunk 2 above)
Comment (zxdong262) [reply src/client/components/setting-panel/tree-list.jsx]: “one click to duplicate it at once”, make sure it do this.

Comment (leonunix) [reply src/client/components/setting-panel/tree-list.jsx]: It is only duplicate bookmark info.  But not save it until user click save button. So i think this is not problem.

Comment (zxdong262) [reply src/client/components/setting-panel/tree-list.jsx]: No, it is not a common practice, think about the duplicate item button in other apps. When user click it, they would expect item duplicated created.

Comment (zxdong262) [reply src/client/components/setting-panel/tree-list.jsx]: Your current solution is not duplicating a bookmark, it is copying item info and pasting into new bookmark form.

Comment (leonunix) [reply src/client/components/setting-panel/tree-list.jsx]: > Your current solution is not duplicating a bookmark, it is copying item info and pasting into new bookmark form.

thank you for suggest. I will improve this. 
But now my macbook is bloom. So i will be improve this in this weekend

Comment (zxdong262) [reply src/client/components/setting-panel/tree-list.jsx]: No problem, take your time.

**Comment by leonunix** [comment]
hi zxdong
I have improve my code
Can you do a review again.
Wait for your suggest

**Comment by zxdong262** [comment]
@leonunix Good job, works as expected, and the code is solid too!

**Pull request merged**